taichi/ir/expr.cpp

```diff
@@ -31,8 +31,8 @@ Expr Expr::operator[](const ExprGroup &indices) const {
     // Allow indexing IndexExpression with ret_type = TensorType
     TI_ASSERT(is_tensor(expr->ret_type));
   } else {
-    TI_ASSERT(is<FieldExpression>() || is<ExternalTensorExpression>() ||
-              is<IdExpression>());
+    TI_ASSERT(is<FieldExpression>() || is<MatrixFieldExpression>() ||
```

Comment:
Would this check disable usage such as `(a @ b)[index]`?

Reply:
Nice catch. Fixed.